awx/main/tasks/jobs.py

```diff
@@ -1153,6 +1155,30 @@ def build_private_data(self, project_update, private_data_dir):
                 private_data['credentials'][credential] = credential.get_input('ssh_key_data', default='')
         return private_data
 
+    def _get_github_app_installation_access_token(self, project_update):
+        jwt_token = jwt.encode(
+            {
+                'iat': int(time.time()),  # Issued at time
+                'exp': int(time.time()) + (10 * 60),  # JWT expiration time (10 minute maximum)
+                'iss': project_update.credential.get_input('github_app_id', default=''),  # GitHub App's identifier
```

Comment:
GH App should be global for the entire AWX. The whole idea is that the users don't need to set up and maintain multiple apps. They just need to click “Install” on GH UI.